programming-languages/README.md

```diff
@@ -56,3 +80,33 @@ It is highly recommended to take the time and get familiar with the standard lib
 
 * [Compiler Explorer](https://godbolt.org/): Online C++ compiler. It lets you compile small code snippets and look at the generated assembly code. It supports different compilers across various architectures ranging from x86 to Arduinos.
 * [JetBrains CLion](https://www.jetbrains.com/clion/): One of the best C++ IDEs. CLion works on CMake making it easy to use with roscpp. 
+
+
+## Java
+
+Java is a powerful programming language that has greatly impacted the development world and is used in all types of technologies. Java can be used to make apps for the web and mobile!
+
+Check out the [official website](https://www.java.com/en/) to find out more!
+
+### Tutorials
+
+* [w3schools](https://www.w3schools.com/java/default.asp): w3schools is a really great resource to learn any part of web development. In this tutorial, you will learn the basics of Java, important programming concepts such as methods and classes, and real life projects you can make yourself!
+
+* [Java Code Academy Course](https://www.codecademy.com/learn/learn-java): Learn Java in this hands on course. In this course, you'll learn important Java concepts such as variables, arrays, strings, and more! By the end of the course, you will have made 6 projects to showcase your new skills!
+
+### Books and Reference Material
+
+* [Java in Easy Steps](https://www.amazon.com/Java-easy-steps-Covers/dp/1840786213): Easy to follow book that explains the basics of Java with fun and interesting graphics.
+
+* [Head First Java](https://www.amazon.com/dp/0596009208/?tag=javamysqlanta-20): Classic book covering Java that is essental for any beginner programmer.
+
+### Channels
+
+* [TheNewBoston Youtube Channel](https://www.youtube.com/channel/UCJbPGzawDH1njbqV-D5HqKw): This youtube channel is jam packed with tons of resources on learning Java. Are you a beginner? Start with the "Java (Beginner) Tutorial" playlist. When you feel you are ready, move onto to the "Java (Intermediate) Tutorial" playlist to learn some more advanced topics!
+
+* [Derek Banas]()https://www.youtube.com/channel/UCwRXb5dUK4cvsHbx-rGzSgw): Channel with crash course videos teaching you the basic of every programming language!
```

Comment:
Remove the redundant ")" from "()https://.....". It is missing up the rich diff

Reply:
Check out my latest commit for the changes! Thanks so much!